graylog2-shared/src/main/java/org/graylog2/shared/buffers/JournallingMessageHandler.java

```diff
@@ -58,14 +62,30 @@ public void onEvent(RawMessageEvent event, long sequence, boolean endOfBatch) th
 
             final Converter converter = new Converter();
             // copy to avoid re-running this all the time
-            final List<Journal.Entry> entries = Lists.newArrayList(transform(batch, converter));
-            final long lastOffset = journal.write(entries);
-            log.debug("Processed batch, wrote {} bytes, last journal offset: {}, signalling reader.",
-                      converter.getBytesWritten(),
-                      lastOffset);
-            journalFilled.release();
+            // Remove all null values returned from the converter (might happen if the Converter throws an exception)
+            final List<Journal.Entry> entries = Lists.newArrayList(filter(transform(batch, converter), notNull()));
```

Comment:
We could save the allocation of the outer-most `ArrayList` if `Journal#write()` took a `Collection<Journal.Entry>` instead of `List<Journal.Entry>`. Since this code is run very often (for each message event in fact), this could make sense and it's a very small change.